lightly/utils/scheduler.py

```diff
@@ -67,6 +66,76 @@ def cosine_schedule(
     return decay
 
 
+def cosine_warmup_schedule(
+    step: int,
+    max_steps: int,
+    start_value: float,
+    end_value: float,
+    warmup_steps: int,
+    warmup_start_value: float,
+    warmup_end_value: Optional[float] = None,
+    period: Optional[int] = None,
+) -> float:
+    """Use cosine decay to gradually modify start_value to reach target end_value.
+
+    Uses linear warmup for the first warmup_steps steps.
+
+    Args:
+        step:
+            Current step number.
+        max_steps:
+            Total number of steps.
+        start_value:
+            Starting value.
+        end_value:
+            Target value.
+        warmup_steps:
+            Number of steps for warmup.
+        warmup_start_value:
+            Starting value for warmup.
+        warmup_end_value:
+            Target value for warmup. Defaults to start_value.
+        period:
+            The number of steps over which the cosine function completes a full cycle.
+            Defaults to max_steps - warmup_steps.
+    Returns:
+        Cosine decay value.
+    """
+    if warmup_steps < 0:
+        raise ValueError("Warmup steps can't be negative")
```

Comment:
Minor: 
```suggestion
        raise ValueError(f"Warmup steps ({warmup_steps}) can't be negative")
```

And you can add it to the other error messages below too